code/obj/machinery/computer/general_air_control.dm

```diff
@@ -567,68 +522,62 @@ Rate: <A href='?src=\ref[src];change_vol=-10'>--</A> <A href='?src=\ref[src];cha
-				signal.data["parameter"] = command
-
-		if (href_list["pressure_adj"] || href_list["pressure_set"])
-			var/pressure = mixer_information["target_pressure"]
-
-			var/amount = 0
-			if (href_list["pressure_adj"])
-				var/diff = text2num_safe(href_list["pressure_adj"])
-				amount = clamp(pressure + diff, 0, MAX_PRESSURE)
-
-			else if (href_list["pressure_set"])
-				var/change = input(usr,"Target Pressure (0 - [MAX_PRESSURE]):", "Enter target pressure", pressure) as num
+		switch (action)
+			if ("toggle_pump")
+				var/status = mixer_information["pump_status"]
+				var/command
+				if (status)
+					if (status == "Offline")
+						command = "power_on"
+					else if (status == "Online")
+						command = "power_off"
+
+				if (command)
+					signal.data["command"] = "toggle_pump"
+					signal.data["parameter"] = command
+
+			if ("pressure_set")
+				var/target_pressure = params["target_pressure"]
 				if ((BOUNDS_DIST(src, usr) > 0 && !issilicon(usr)) || !isliving(usr) || iswraith(usr) || isintangible(usr))
 					return 0
 				if (is_incapacitated(usr) || usr.restrained())
 					return 0
 				if (!src.allowed(usr))
 					boutput(usr, SPAN_ALERT("Access denied!"))
 					return 0
-				if (!isnum_safe(change))
+				if (!isnum_safe(target_pressure))
 					return 0
 
-				amount = clamp(change, 0, MAX_PRESSURE)
-
-			signal.data["command"] = "set_pressure"
-			signal.data["parameter"] = num2text(amount)
+				var/amount = clamp(target_pressure, 0, MAX_PRESSURE)
 
-		if (href_list["ratio"])
-			var/amount = text2num_safe(href_list["ratio"])
-			var/volume_rate = mixer_information["i1trans"]
+				signal.data["command"] = "set_pressure"
+				signal.data["parameter"] = num2text(amount)
 
-			signal.data["command"] = "set_ratio"
-			signal.data["parameter"] = num2text(volume_rate + amount)
+			if ("ratio")
+				signal.data["command"] = "set_ratio"
+				signal.data["parameter"] = params["ratio"]
 
-			if (src.id == "pmix_control")
-				if (((src.last_change + src.message_delay) <= world.time))
-					src.last_change = world.time
-					logTheThing(LOG_STATION, usr, "has just edited the plasma mixer at [log_loc(src)].")
-					message_admins("[key_name(usr)] has just edited the plasma mixer at at [log_loc(src)].")
+				if (src.id == "pmix_control")
+					if (((src.last_change + src.message_delay) <= world.time))
+						src.last_change = world.time
+						logTheThing(LOG_STATION, usr, "has just edited the plasma mixer at [log_loc(src)].")
+						message_admins("[key_name(usr)] has just edited the plasma mixer at at [log_loc(src)].")
 
-		if (href_list["refresh_status"])
-			signal.data["status"] = 1
+			if ("refresh_status")
+				signal.data["status"] = 1
 
 		if (signal)
 			SEND_SIGNAL(src, COMSIG_MOVABLE_POST_RADIO_PACKET, signal)
-#undef MAX_PRESSURE
+			. = TRUE
+
+	ui_status(mob/user, datum/ui_state/state)
+		. = ..()
 
-#undef _GET_SIGNAL_GAS
-#undef GET_SIGNAL_MIXTURE
+		if (!src.allowed(user) && . >= UI_UPDATE)
+			. = UI_UPDATE
```

Comment:
What's this trying to achieve? Capping the UI to "update" if the user is not "allowed", sure, but what does that mean and why is it unique to gas mixers? 🤔

Reply:
People without engineering/toxins access are allowed to view the UI but aren't allowed to change settings or turn it on/off. Thinking about it a Dimmer over the pump section works much better at telling the user they're disallowed so I went with that.